artifactory/utils/dotnet/solution/solution.go

```diff
@@ -75,6 +94,32 @@ func getModuleId(customModuleID, projectName string) string {
 	return projectName
 }
 
+// Populate requested by field for the input dependencies.
+// parentDependency - The parent dependency
+// dependenciesMap  - The input dependencies map
+// childrenMap      - Map from dependency ID to children IDs
+func populateRequestedBy(parentDependency buildinfo.Dependency, dependenciesMap map[string]*buildinfo.Dependency, childrenMap map[string][]string) {
+	if parentDependency.NodeHasLoop() {
```

Comment:
maybe you should move it inside the children forloop as [I did in go](https://github.com/jfrog/build-info-go/blob/83db6708f2b24369d93982e264de8e9286541d14/build/golang.go#L172)

So if we catch a loop we don't add it to the tree and continue to the next child